tests/unit/test_markdown.py

```diff
@@ -225,6 +227,22 @@ def should_generate_env_prefix_and_nested_delimiter(runner: CliRunner, mocker: M
         assert "`prefix_sub_model__nested`" in actual_string
         assert "`prefix_sub_model__deep__leaf`" in actual_string
 
+    @staticmethod
+    def should_generate_settings_with_settings_sub_model(runner: CliRunner, mocker: MockerFixture):
+        actual_string = run_app_with_settings(mocker, runner, SettingsWithSettingsSubModel)
+
+        assert "`prefix_direct`" in actual_string
+        assert "`prefix_sub_model_nested`" in actual_string
+        assert "`prefix_sub_model_sub_model_2__leaf`" in actual_string
```

Comment:
Please add also the negative examples of what should not be rendered. In the test below as well.

Reply:
Added. LMK if that's not what you had in mind.